src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -441,6 +447,11 @@ public void Enqueue(IThreadPoolWorkItem callback, bool forceGlobal)
             if (null != tl)
             {
                 tl.workStealingQueue.LocalPush(callback);
+
+                if (Volatile.Read(ref WorkStealingQueueList.wsqActive) != WorkStealingQueueList.WsqNowActive)
```

Comment:
Sorry have to iterate on this again. I don't think the volatile read here prevents the read from being reordered before the write in LocalPush's fast path because the barrier would be after the read and before LocalPush's write. Is there a possibility that the reordered read prevents the write from occurring even though a work item was pushed?